hashicorp/boundary-ui
# fix url when user tries to access resource with incorrect url

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-11282)

## Description


Add redirect route hook to routes that retrieve a singular resource. The redirect hook verifies that the parent resource id in the url path matches the parent resource id existing on that singular resource. For example, the redirect hook in a account.js route file verifies that the `auth_method_id` in an account model matches the auth-method id present in the url. If the ids do not match the url is corrected.

Add beforeModel route hook to routes that create a new resource. Redirect user to parent route if user does not have create authorized collection action.

## How to Test


Acceptance tests were added.
To verify use a url with an incorrect dynamic segment (resource id) and watch redirect to correct url.
Ex: use url with incorrect scope for an auth-method.

## Checklist:
